js/tab.js

```diff
@@ -171,6 +180,32 @@ define([
     return syntaxValue;
   }
   
+  Tab.prototype.getWorker = function() {
+    return new Promise((resolve, reject) => {
+      // if worker is off, return null
+      if (!Settings.get("user").useWorker) return resolve(null);
+      
+      var checkLimit = 20;
+      var findWorker = () => {
+        // if worker was turned while we were waiting on it, return null
+        if (!Settings.get("user").useWorker) return resolve(null);
+      
+        // return the worker if we found it
+        if (this.$worker) {
+          return resolve(this.$worker);
+        }
+        
+        // see if we've checked too many times
+        if (checkLimit-- < 0) return reject(new Error('Could not find worker.'));
+        
+        // otherwise, go to sleep and then check again
+        setTimeout(findWorker, 100);
```

Comment:
So this is just me logging out some function calls and the state of `this.$worker` in tab.js. Both calls logging `this.$worker` are in detectSyntax, one in a `setTimeout`.

```
detectSyntax
render
detectSyntax
this.$worker: undefined
this.$worker: undefined
render
this.$worker 100ms later: null
this.$worker 100ms later: null
setFile
setPath
detectSyntax
this.$worker: undefined
render
this.$worker 100ms later: { WorkerClient }
```

It seems like there is some async-ness going on before a Tab has access to its worker.